ansible/library/monit_process.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+import multiprocessing
+import operator
+try:
+    import psutil
+except ImportError:
+    HAS_PSUTIL = False
+else:
+    HAS_PSUTIL = True
+import re
+import time
+from ansible.module_utils.basic import AnsibleModule
+from collections import namedtuple
+
+DOCUMENTATION = """
+module: monit_process
+short_description: retrieve process cpu and memory usage
+description:
+    - Retrieve process cpu and memory usage for a given period
+version_added: "2.8"
+options:
+    delay_interval:
+        description:
+            - Specify the delay interval between each polling(in seconds)
+        required: false
+        default: 5
+    iterations:
+        description:
+            - Specify the maximum number of polling iterations
+        required: false
+        default: 12
+"""
+
+MonitResult = namedtuple('MonitResult', ['processes', 'memory'])
+
+
+def monit_process(module, interval, iterations):
+    """Retrieve process statistics."""
+
+    def _monit_processes_with_top(module, interval, iterations):
+        cmd = 'top -d {interval} -n {iterations} -b'.format(
+            interval=interval, iterations=(iterations + 1)
+            )
+        rc, stdout, _ = module.run_command(args=cmd)
+
+        monit_results = []
+        proc_section = False
+        mem_re = re.compile(
+            (r"^KiB Mem\s+:\s+(?P<total>\d+)\s+total,\s+(?P<free>\d+)"
+             r"\s+free,\s+(?P<used>\d+)\s+used,\s+\d+\s+buff/cache$")
+        )
```

Comment:
The output of top is different in different platforms. On dx010-4, I got following output
```
root@str-dx010-acs-4:~# top -d 12 -n 1 -b
top - 09:28:21 up 18:32,  2 users,  load average: 2.19, 2.41, 2.28
Tasks: 264 total,   1 running, 233 sleeping,   0 stopped,  30 zombie
%Cpu(s): 14.7 us, 30.7 sy,  0.0 ni, 54.7 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
MiB Mem :   3923.4 total,    504.7 free,   2143.7 used,   1275.1 buff/cache
MiB Swap:      0.0 total,      0.0 free,      0.0 used.   1481.0 avail Mem 
```
The Mem is in **MiB** rather that **KiB**. Can we use other method to retrieve mem usage (like free)?

Reply:
Updated, using `-E` to enforce uniform output size unit.